src/xrGame/space_restrictor.cpp

```diff
@@ -100,7 +100,14 @@ void CSpaceRestrictor::net_Destroy()
 bool CSpaceRestrictor::inside(const Fsphere& sphere) const
 {
     if (!actual())
-        prepare();
+    {
+        static std::mutex prepareMutex;
+        std::lock_guard lock(prepareMutex);
+
+        // Double-checked locking
+        if (!actual())
+            prepare();
+    }
```

Comment:
I believe it was only non-concurrent part for parallel for in `CSpaceRestrictionShape::fill_shape`. Since it is called only once for a sequence of `CSpaceRestrictor::inside` calls, maybe it would be better to extract this to reduce contention